### PR TITLE
Environment_Engine: Created new methods for IsShade

### LINKED ISSUE #2792: Environment_Engine: References to PanelType.Shade should include TranslucentShade
#### Broken rules:

Some methods, such as `Query/ExposedToSun` made use of comparisons to the `PanelType.Shade` enum value. This ignores the new `TranslucentShade` option brought in in 5.1.

#### Suggestions to restore compliance:

Go through the engine code base and update any instance of `PanelType.Shade` to also incorporate `PanelType.TranslucentShade`.

Suggestion from @FraserGreenroyd :

Make a new query method for whether a panel is a shading panel - `Query/IsShade` which can check for all different types of shade enum values to update things on later.

Actions to resolve from @FraserGreenroyd :

1. Create 2 new Query methods called `IsShade`, one taking in a `Panel` object, and one taking in a `PanelType` object.
2. In the first new method taking in a `PanelType` object, return a `bool` for whether the `PanelType` is one of the accepted shade options (currently just `Shade` and `TranslucentShade`.
3. In the second new method taking in a `Panel` object, return a `bool` for whether the `Panel.Type.IsShade()` - call the new `IsShade` method on the panel type property.
4. Go through the engine for any instances of `PanelType.Shade` - replace with queries to the new `IsShade` method.

### PR DESCRIPTION
**Pull request by robinflyman**
Closes #2792 

Added IsShade methods according to Frasers suggestion.

### CODE REVIEW

**Comment by robinflyman** [comment]
@BHoMBot check versioning

File: Environment_Engine/Modify/SetShadePanels.cs [hunk 1]
```diff
@@ -42,15 +42,22 @@ public static partial class Modify
 
         [Description("Returns the shade panels represented by Environment Panels with no adj spaces and fixes PanelType")]
         [Input("panels", "A collection of Environment Panels")]
+        [Input("panelType", "A panelType enum.")]
```
Comment (FraserGreenroyd) [inline Environment_Engine/Modify/SetShadePanels.cs:45]:
```suggestion
        [Input("shadeType", "The type of shade to assign to the shade panels.")]
```

File: Environment_Engine/Modify/SetShadePanels.cs [hunk 2]
```diff
@@ -42,15 +42,22 @@ public static partial class Modify
...
         [Output("shadePanels", "BHoM Environment panel representing the shade")]
-        public static List<Panel> SetShadePanels(this List<Panel> panels)
+        public static List<Panel> SetShadePanels(this List<Panel> panels, PanelType panelType = PanelType.Shade)
```
Comment (FraserGreenroyd) [inline Environment_Engine/Modify/SetShadePanels.cs:47]:
```suggestion
        public static List<Panel> SetShadePanels(this List<Panel> panels, PanelType shadeType = PanelType.Shade)
```

File: Environment_Engine/Modify/SetShadePanels.cs [hunk 3]
```diff
@@ -42,15 +42,22 @@ public static partial class Modify
...
         {
+            if (!panelType.IsShade())
```
Comment (FraserGreenroyd) [inline Environment_Engine/Modify/SetShadePanels.cs:49]:
```suggestion
            if (!shadeType.IsShade())
```

File: Environment_Engine/Modify/SetShadePanels.cs [hunk 4]
```diff
@@ -42,15 +42,22 @@ public static partial class Modify
...
+                return null;
+            }
+
             //Find the panel(s) without connected spaces and set as shade
             List<Panel> shadePanels = new List<Panel>(panels.Select(x => x.DeepClone<Panel>()).ToList());
             foreach (Panel panel in shadePanels)
             {
                 if (panel.ConnectedSpaces.Where(x => x != "-1").ToList().Count == 0)
-                    panel.Type = PanelType.Shade;
+                    panel.Type = panelType;
```
Comment (FraserGreenroyd) [inline Environment_Engine/Modify/SetShadePanels.cs:60]:
```suggestion
                    panel.Type = shadeType;
```

File: Environment_Engine/Query/IsShade.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+using BH.oM.Environment.Elements;
+
+namespace BH.Engine.Environment
+{
+    public static partial class Query
+    {
+        [Description("Method will check what type the Paneltype is, and return false if it's not a shade-type.")]
+        [Input("panelType", "PanelType to determine if it's a shade-type or not.")]
+        [Output("Boolean", "Will return true if the paneltype is a shade-type.")]
```
Comment (FraserGreenroyd) [inline Environment_Engine/Query/IsShade.cs:37]:
```suggestion
        [Output("isShade", "Will return true if the provided panel type is a shade-type.")]
```

File: Environment_Engine/Query/IsShade.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
```
Comment (FraserGreenroyd) [inline Environment_Engine/Query/IsShade.cs:38]:
```suggestion
```

File: Environment_Engine/Query/IsShade.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        {
+            List<PanelType> shadeTypes = new List<PanelType>() {
+                PanelType.Shade,
+                PanelType.TranslucentShade,
+            };
+
+            return shadeTypes.Contains(panelType);
+        }
+
+        [Description("Method will check what type the Paneltype is, and return false if it's not a shade-type.")]
```
Comment (FraserGreenroyd) [inline Environment_Engine/Query/IsShade.cs:49]:
```suggestion
        [Description("Ascertain whether a given panel is a shading panel.")]
```

File: Environment_Engine/Query/IsShade.cs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        [Input("Panel", "Panel to determine if it's a shade-type or not.")]
```
Comment (FraserGreenroyd) [inline Environment_Engine/Query/IsShade.cs:50]:
```suggestion
        [Input("panel", "Panel to determine if it's a shade-type or not.")]
```

File: Environment_Engine/Query/IsShade.cs [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        [Output("Boolean", "Will return true if the paneltype is a shade-type.")]
```
Comment (FraserGreenroyd) [inline Environment_Engine/Query/IsShade.cs:51]:
```suggestion
        [Output("isShade", "Will return true if the panel is a shade-type.")]
```

File: Environment_Engine/Query/IsShade.cs [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
```
Comment (FraserGreenroyd) [inline Environment_Engine/Query/IsShade.cs:52]:
```suggestion
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
In addition to the versioning - see my code suggestions.

You need to delete `Geometry_Engine/Geometry_Engine - Backup.csproj` and `Graphics_Engine/Graphics_Engine - Backup.csproj` and commit the deletions as neither of those files are required.

**Comment by robinflyman** [comment]
@BHoMBot check versioning

File: Environment_Engine/Modify/SetShadePanels.cs [hunk 11]
```diff
@@ -40,17 +40,24 @@ public static partial class Modify
...
 
-        [Description("Returns the shade panels represented by Environment Panels with no adj spaces and fixes PanelType")]
-        [Input("panels", "A collection of Environment Panels")]
-        [Output("shadePanels", "BHoM Environment panel representing the shade")]
-        public static List<Panel> SetShadePanels(this List<Panel> panels)
+        [Description("Returns the shade panels represented by Environment Panels with no adj spaces and fixes PanelType.")]
+        [Input("panels", "A collection of Environment Panels.")]
+        [Input("shadeType", "The type of shade to assign to the shade panels.")]
+        [Output("shadePanels", "BHoM Environment panel representing the shade.")]
+        public static List<Panel> SetShadePanels(this List<Panel> panels, PanelType shadeType = PanelType.Shade)
```
Comment (FraserGreenroyd) [inline Environment_Engine/Modify/SetShadePanels.cs:47]:
```suggestion
        [Output("shadePanels", "BHoM Environment panel representing the shade.")]
        [PreviousVersion("5.2", "BH.Engine.Environment.Modify.SetShadePanels(System.Collections.Generic.List<BH.oM.Environment.Elements.Panel>)")]
        public static List<Panel> SetShadePanels(this List<Panel> panels, PanelType shadeType = PanelType.Shade)
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

File: Environment_Engine/Query/IsShade.cs [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            List<PanelType> shadeTypes = new List<PanelType>() {
+                PanelType.Shade,
+                PanelType.TranslucentShade,
+            };
+
+            return shadeTypes.Contains(panelType);
+        }
+
+        [Description("Acertain whether a given panel is a shading panel.")]
```
Comment (jamesramsden-bh) [inline Environment_Engine/Query/IsShade.cs:48]: Ascertain

File: Environment_Engine/Query/IsShade.cs [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+using BH.oM.Base.Attributes;
+using System.ComponentModel;
+
+using BH.oM.Environment.Elements;
+
+namespace BH.Engine.Environment
+{
+    public static partial class Query
+    {
+        [Description("Acertain whether a given panel type is an acceptable shade type.")]
```
Comment (jamesramsden-bh) [inline Environment_Engine/Query/IsShade.cs:35]: Ascertain

**Review by jamesramsden-bh — CHANGES_REQUESTED** [review]
Couple of spelling errors noticed. Haven't yet tested functionality.

File: Environment_Engine/Query/IsShade.cs (hunk 13 above)
Comment (jamesramsden-bh) [reply Environment_Engine/Query/IsShade.cs:35]: Actually, I probably would reword this. Something like "Ascertain whether a given PanelType is a type of shade."

**Review by jamesramsden-bh — APPROVED** [review]
Checked a panel with different panel types as below. isShade returns the correct value. 

image

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning
@BHoMBot check installer

File: Environment_Engine/Modify/SetShadePanels.cs (hunk 11 above)
Comment (FraserGreenroyd) [inline Environment_Engine/Modify/SetShadePanels.cs:47]:
```suggestion
        [PreviousVersion("5.3", "BH.Engine.Environment.Modify.SetShadePanels(System.Collections.Generic.List<BH.oM.Environment.Elements.Panel>)")]
        public static List<Panel> SetShadePanels(this List<Panel> panels, PanelType shadeType = PanelType.Shade)
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Review by FraserGreenroyd — APPROVED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am authorising dispensation to be granted on check ref. 7524112114
